optimum/intel/openvino/modeling_sam.py

```diff
@@ -0,0 +1,427 @@
+                standard cache should not be used.
+            vision_encoder_file_name(`str`, *optional*):
+                The vision encoder model file name. Overwrites the default file name openvino_vision_encoder.xml and allows one to
+                load the vision encoder model with a different name.
+            prompt_encoder_mask_decoder_file_name(`str`, *optional*):
+                The prompt encdoer and mask decoder model file name overwriting the default file name
+                openvino_prompt_encoder_mask_decoder.xml, allowing to load the decoder model with a different name.
+            local_files_only(`bool`, *optional*, defaults to `False`):
+                Whether or not to only look at local files (i.e., do not try to download the model).
+        """
+        if use_auth_token is not None:
+            warnings.warn(
+                "The `use_auth_token` argument is deprecated and will be removed soon. Please use the `token` argument instead.",
+                FutureWarning,
+            )
+            if token is not None:
+                raise ValueError("You cannot use both `use_auth_token` and `token` arguments at the same time.")
+            token = use_auth_token
+
+        from_onnx = kwargs.get("from_onnx", False)
+
+        default_vision_encoder_file_name = (
+            ONNX_VISION_ENCODER_MODEL_NAME if from_onnx else OV_VISION_ENCODER_MODEL_NAME
+        )
+        default_prompt_encoder_mask_decoder_file_name = (
+            ONNX_PROMPT_ENCODER_MASK_DECODER_MODEL_NAME if from_onnx else OV_PROMPT_ENCODER_MASK_DECODER_MODEL_NAME
+        )
+        vision_encoder_file_name = vision_encoder_file_name or default_vision_encoder_file_name
+        prompt_encoder_mask_decoder_file_name = (
+            prompt_encoder_mask_decoder_file_name or default_prompt_encoder_mask_decoder_file_name
+        )
+
+        model_file_names = {
+            "vision_encoder_model": vision_encoder_file_name,
+            "prompt_encoder_mask_decoder_model": prompt_encoder_mask_decoder_file_name,
+        }
+
+        if not from_onnx:
+            model_file_names.update(
+                {
+                    "vision_encoder_model_bin": vision_encoder_file_name.replace(".xml", ".bin"),
+                    "prompt_encoder_mask_decoder_model_bin": prompt_encoder_mask_decoder_file_name.replace(
+                        ".xml", ".bin"
+                    ),
+                }
+            )
+
+        compile_only = kwargs.get("compile_only", False)
+        if os.path.isdir(model_id):
+            # Load model from a local directory
+            model_save_dir = Path(model_id)
+            file_names = {k: os.path.join(model_id, model_file_names[k]) for k in model_file_names}
+        else:
+            file_names = {}
+            for name, file_name in model_file_names.items():
+                model_cache_path = hf_hub_download(
```

Comment:
please use the `_cached_file` method whenever possible, it abstracts the from_onnx logic and It is more robust against CI rate limiting

Reply:
https://github.com/huggingface/optimum-intel/blob/main/optimum/intel/openvino/modeling_base.py#L522

Reply:
reused, PR was opened before your changes